### PR TITLE
Add BigTableIO Stress test

### PR DESCRIPTION
**Pull request by akashorabek**
This pull request introduces stress tests for BigTableIO, designed to assess the performance under various conditions. The stress tests simulate dynamic load increases and evaluate the behavior of BigTableIO.

Changes:

- Added stress tests for BigTableIO.
- Implemented dynamic load increases over time to simulate varying workloads.
- Added support for exporting metrics to InfluxDB or BigQuery based on the configuration parameter.
- Added ability for user to configure Dataflow version in KafkaIO Stress test
- Refactored the code of BigQueryIO and KafkaIO stress tests.

[Writing job example](https://console.cloud.google.com/dataflow/jobs/us-central1/2024-03-14_01_04_09-6829471925970253865;step=Write%20to%20BigTable;mainTab=JOB_GRAPH;bottomTab=WORKER_LOGS;logsSeverity=WARNING;graphView=0?project=apache-beam-testing&pageState=(%22dfTime%22:(%22l%22:%22dfJobMaxTime%22)))

[Reading job example](https://console.cloud.google.com/dataflow/jobs/us-central1/2024-03-14_03_46_28-3739931199332933782;step=Read%20from%20BigTable;graphView=0?project=apache-beam-testing&pageState=(%22dfTime%22:(%22l%22:%22dfJobMaxTime%22)))

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Review by Abacn — COMMENTED** [review]
Thanks for the change, especially the systemmatic refactor work. Just had one more comment regarding the common method below

File: it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/IOStressTestBase.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        }
+      }
+      for (int i = 0; i < multiplier; i++) {
+        outputReceiver.output(element);
+      }
+    }
+  }
+
+  /** Exports test metrics to InfluxDB or BigQuery depending on the configuration. */
+  protected void exportMetrics(
```
[the remark this answers is not in the document]
Comment (akashorabek) [reply it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/IOStressTestBase.java]: Done

**Comment by akashorabek** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by akashorabek** [comment]
Run Kotlin_Examples PreCommit

**Comment by akashorabek** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by akashorabek** [comment]
Run Kotlin_Examples PreCommit

**Comment by akashorabek** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by akashorabek** [comment]
Run Kotlin_Examples PreCommit

**Comment by akashorabek** [comment]
Run Kotlin_Examples PreCommit

**Comment by akashorabek** [comment]
Run Kotlin_Examples PreCommit

**Comment by akashorabek** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by akashorabek** [comment]
Run Kotlin_Examples PreCommit

**Comment by akashorabek** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by akashorabek** [comment]
Run Java_Examples_Dataflow PreCommit

**Review by Abacn — APPROVED** [review]
Thank you!

**Pull request merged**